src/tactic/linear_combination.lean

```diff
@@ -415,10 +435,16 @@ by linear_combination 3 * h a b + hqc
 meta def _root_.tactic.interactive.linear_combination
   (input : parse (as_linear_combo ff [] <$> texpr)?)
   (_ : parse (tk "with")?)
+  (exponent : parse (prod.mk <$> ident <*> small_nat)?)
```

Comment:
This results in the slightly unhelpful help text,
```
linear_combination expr? with? (id n)? linear_combo.linear_combination_config?
```

I think either:
* `exponent` should appear in this help text
* we should make `exponent` a field of `linear_combination_config`

Reply:
I'd lean towards the second; `linear_combination _ with { exponent := 2 }` is much easier to extend later, and doesn't involve any surprising new syntax.

Reply:
> I'd lean towards the second; `linear_combination _ with { exponent := 2 }` is much easier to extend later, doesn't involve any surprising new syntax, and has default values built in.

I'm not opposed, but there's another `linear_combination` feature under discussion that wouldn't fit this pattern. We're scheming up syntax for a "contraposed" version, something like `linear_combination 3*⊢ with target zero_ne_one` that would work on disequality goals and a single disequality hypothesis/proof term. I don't think we can take an arbitrary proof of an arbitrary proposition in the config structure with convenient syntax and a default value.

So that would leave us at `linear_combination _ with target zero_ne_one {exponent := 2}` which is a little funny.

Reply:
What would your proposed spelling me for both options simultaneously?

Reply:
At a glance, "linear_combination _ with target x" feels a bit like `simpa _ using x`

Reply:
@eric-wieser I've taken your suggestion and reimplemented this as a field of the config object!